tests/test_client.py

```diff
@@ -0,0 +1,175 @@
+from tap_square.client import SquareClient
+import singer
+import uuid
+import os
+import random
+
+LOGGER = singer.get_logger()
+
+
+typesToKeyMap = {
+    'ITEM': 'item_data',
+    'CATEGORY': 'category_data',
+    'DISCOUNT': 'discount_data',
+    'TAX': 'tax_data'
+}
+
+
+class TestClient(SquareClient):
+
+    stream_to_data_schema = {
+        'items': {'type': 'ITEM',
+                  'item_data': {'name': 'tap_tester_item_data'}},
+        'categories': {'type': 'CATEGORY',
+                       'category_data': {'name': 'tap_tester_category_data'}},
+        'discounts': {'type': 'DISCOUNT',
+                      'discount_data': {'name': 'tap_tester_discount_data',
+                                        'discount_type': 'FIXED_AMOUNT',
+                                        'amount_money': {'amount': 34500,
+                                                         'currency': 'USD'}}},
+        'taxes': {'type': 'TAX',
+                  'tax_data': {'name': 'tap_tester_tax_data'}},
+    }
+
+    def __init__(self):
+        config = {
+            'refresh_token': os.getenv('TAP_SQUARE_REFRESH_TOKEN'),
+            'client_id': os.getenv('TAP_SQUARE_APPLICATION_ID'),
+            'client_secret': os.getenv('TAP_SQUARE_APPLICATION_SECRET'),
+            'sandbox': 'true'
+        }
+
+        super().__init__(config)
+
+    def get_all(self, stream, start_date=None, end_date=None):
+        if stream == 'items':
+            return [obj for page, _ in self.get_catalog('ITEM', start_date, None) for obj in page]
+        elif stream == 'categories':
+            return [obj for page, _ in self.get_catalog('CATEGORY', start_date, None) for obj in page]
+        elif stream == 'discounts':
+            return [obj for page, _ in self.get_catalog('DISCOUNT', start_date, None) for obj in page]
+        elif stream == 'taxes':
+            return [obj for page, _ in self.get_catalog('TAX', start_date, None) for obj in page]
+        elif stream == 'employees':
+            return [obj for page, _ in self.get_employees(None) for obj in page]
+        elif stream == 'locations':
+            return [obj for page, _ in self.get_locations() for obj in page]
+        else:
+            raise NotImplementedError
```

Comment:
I don't quite understand what is happening here.

Reply:
For the stream you pass in, call the `get` function from the tap. The `get`s returns 1 or more pages, so we loop over every page and grab every record

Reply:
Interesting. In adroll the test client defines it's own GET function. I am not sure if separating them adds value or not though. My initial reaction to this is it is bad that we are using the method that we are testing to set the expectations. But I do not know if rewriting the function in a different way is any better.